### ISSUE TITLE
[upgrade test failure] Garbage collector should support cascading deletion of custom resources

### ISSUE THREAD

**Issue by ericchiang** [kubernetes/kubernetes#50793]
/cc @kubernetes/sig-api-machinery-bugs 

This test is failing on several of the 1.7 to master upgrade tests

https://k8s-testgrid.appspot.com/master-upgrade#gke-cvm-1.7-cvm-master-upgrade-cluster-new
https://k8s-testgrid.appspot.com/master-upgrade#gke-cvm-1.7-gci-master-upgrade-cluster-new
https://k8s-testgrid.appspot.com/master-upgrade#gke-gci-1.7-cvm-master-upgrade-cluster-new
https://k8s-testgrid.appspot.com/master-upgrade#gke-gci-1.7-gci-master-upgrade-cluster-new

Example failures

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-cvm-new-cvm-master-upgrade-cluster-new/133#sig-api-machinery-garbage-collector-should-support-cascading-deletion-of-custom-resources
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-gci-new-gci-master-upgrade-cluster-new/135#sig-api-machinery-garbage-collector-should-support-cascading-deletion-of-custom-resources

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apimachinery/garbage_collector.go:936
Aug 16 13:36:37.288: failed waiting for dependent resource "dependentf7vbw" to be deleted
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apimachinery/garbage_collector.go:924
```

cc @kubernetes/release-maintainers @mbohlool

**Comment by caesarxuchao** [comment]
cc @ironcladlou

**Comment by ironcladlou** [comment]
Already tracked (https://github.com/kubernetes/kubernetes/issues/34120#issuecomment-322503808). I've been unable to get access to the logs for these jobs, they may not even exist if the artifact capture mechanism is breaking.

**Comment by caesarxuchao** [comment]
Probably it's because they are gke tets, so there is no master log.

**Comment by mbohlool** [comment]
Friendly ping. this is a release blocking test failure.

**Comment by ironcladlou** [comment]
@mbohlool No useful artifacts (master/apiserver/controller logs) are being captured (or exposed) to me by Gubernator, and I don't have any access to the Google CI infrastructure, so I'm not sure what I can do to help here. I asked for and failed to get any assistance understanding why there aren't any artifacts weeks ago. Who can diagnose the CI artifact issue?

**Comment by ericchiang** [comment]
@ironcladlou we've tracked the log issue down to a possible scp mis-configuration https://github.com/kubernetes/test-infra/issues/4323

Will update this issue once that's fixed. cc @mbohlool

**Comment by mbohlool** [comment]
@caesarxuchao Can you take a look at this?

**Comment by caesarxuchao** [comment]
Taking a look now.

**Comment by caesarxuchao** [comment]
It seems to be somehow fixed recently: https://k8s-testgrid.appspot.com/master-upgrade#gke-gci-1.7-gci-master-upgrade-cluster-new

**Comment by caesarxuchao** [comment]
The diff when the test started to pass in https://k8s-testgrid.appspot.com/master-upgrade#gke-gci-1.7-gci-master-upgrade-cluster-new. They don't seem to be related to GC or CRD.

Diff in test-infra: doesn't seem to be related
```


$ $  git log 54632f94a..11274d7f7
commit 11274d7f753c3d0ba7c34d72ee54847dba2483e1
Merge: f0a479c051 bd92bac60c
Author: Sen Lu <senlu@google.com>
Date:   Wed Sep 6 14:52:51 2017 -0700

    Merge pull request #4423 from luxas/kubeadm_older_controlplane_master
    
    Deploy the stable version of the control plane with kubeadm from master

commit f0a479c051c56f84e54c70d4795b2c4a8d80d7b9
Merge: 54632f94a3 048dfa447b
Author: Sen Lu <senlu@google.com>
Date:   Wed Sep 6 14:52:08 2017 -0700

    Merge pull request #4424 from luxas/kubeadm_parallel
    
    Run kubeadm e2e tests in parallel

commit bd92bac60cc1f3a6c738fc89f1fdff754a9613f5
Author: Lucas Käldström <lucas@weave.works>
Date:   Thu Sep 7 00:15:24 2017 +0300

    Deploy the stable version of the control plane with kubeadm from master

commit 048dfa447b06b343e2e590cb3295634e4ecf93b6
Author: Lucas Käldström <lucas@weave.works>
Date:   Thu Sep 7 00:13:49 2017 +0300

    Run kubeadm e2e tests in parallel
```

Diff in the master: doesn't seem to be related either.
v1.9.0-alpha.0.391+440884be9ae54a..v1.9.0-alpha.0.395+50c633182ecf18
```
$ git log 440884be9ae54a..50c633182ecf18
commit 50c633182ecf188a3be156b818a9af447576ee4d
Merge: dc9859754a 2bea4f7dd1
Author: Kubernetes Submit Queue <k8s-merge-robot@users.noreply.github.com>
Date:   Wed Sep 6 14:43:01 2017 -0700

    Merge pull request #51987 from nicksardo/gce-fix-updowntests
    
    Automatic merge from submit-queue (batch tested with PRs 51839, 51987)
    
    GCE: Fall back to network if subnet is unknown
    
    Fixes #51981
    
    /assign @bowei
    
    **Release note**:
    ```release-note
    NONE
    ```

commit dc9859754a29e4477c7798f04d03f4208eb892b7
Merge: 440884be9a 3ebd68abf6
Author: Kubernetes Submit Queue <k8s-merge-robot@users.noreply.github.com>
Date:   Wed Sep 6 14:42:58 2017 -0700

    Merge pull request #51839 from jennybuckley/master
    
    Automatic merge from submit-queue (batch tested with PRs 51839, 51987)
    
    Disable rbac/v1alpha1, settings/v1alpha1, and scheduling/v1alpha1 by default
    
    **What this PR does / why we need it**: Disables alpha features which were previously enabled by default. Also changes tests which relied on these alpha features being enabled by default.
    
    **Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #47691
    
    **Special notes for your reviewer**:
    
    **Release note**:
    
    ```release-note
    Fixed a bug where some alpha features were enabled by default.

commit 2bea4f7dd1f468ab5e1b630f24acdf7e748077fc
Author: Nick Sardo <nicksardo@google.com>
Date:   Wed Sep 6 11:59:47 2017 -0700

    Fall back to network if subnet is unknown

commit 3ebd68abf664a1cd38f44adda07d0b34787ce2ed
Author: jennybuckley <jennybuckley@google.com>
Date:   Tue Sep 5 14:10:24 2017 -0700

    Disable rbac/v1alpha1 settings/v1alpha1 scheduling/v1alpha1
```

**Comment by caesarxuchao** [comment]
The logs are still missing in these tests, so it's very difficult to debug. I'll stop spending time on them before https://github.com/kubernetes/test-infra/issues/4323 is fixed.

My hunch is this is not a release blocker, since it's only failing in one particular upgrade test setup. Let's review the test results again next week.

**Comment by ericchiang** [comment]
This appears to be green now on all master upgrade blocking tests. Closing.
